module/localisation/RobotParticleLocalisation/src/RobotModel.hpp

```diff
@@ -78,12 +78,11 @@ namespace module::localisation {
             return state;
         }
 
-        Eigen::Matrix<Scalar, Eigen::Dynamic, 1> predict(
-            const StateVec& state,
-            const Eigen::Matrix<Scalar, Eigen::Dynamic, 1>& actual_position,
-            const Eigen::Matrix<Scalar, 4, 4>& Hcw,
-            const message::vision::Goal::MeasurementType& type,
-            const message::support::FieldDescription& fd) {
+        Eigen::Matrix<Scalar, Eigen::Dynamic, 1> predict(const StateVec& state,
+                                                         const Eigen::Matrix<Scalar, 3, 1>& actual_position,
+                                                         const Eigen::Matrix<Scalar, 4, 4>& Hcw,
```

Comment:
Should this be affine 3d?

Reply:
It could be, but we would have to change where it's called. When it's called, we're passing it straight from the message which contains it, which is converted as a `Matrix4[Scalar]`. So to do that, we would have to wrap it in an `Affine3d` ctor at each call

Reply:
Where is it called, so I can have a geeze?

Reply:
`UKF.hpp` and `ParticleFilter.hpp`, in `measure(..)`

Reply:
Pretty sure they have a converter so it'd probably just work it out

Reply:
They don't. It needs to be explicit, I am pretty sure